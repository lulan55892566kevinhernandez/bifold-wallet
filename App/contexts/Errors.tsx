import React, { useState, useContext } from 'react'
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native'

import AppStyles from '../../assets/styles'

const ErrorsContext = React.createContext({})

interface IErrorDialog {
  path: string
  text: string
}

function ErrorDialog(props: IErrorDialog) {
  const errors = useContext<any>(ErrorsContext)

  return (
    <>
      <View style={styles.errorView}>
        <View style={[AppStyles.messageBox, styles.errorMessage]}>
          <Text style={[AppStyles.h2, AppStyles.textSecondary, AppStyles.textBold]}>ERROR{'\n'}</Text>
          <Text style={[AppStyles.h3, AppStyles.textSecondary]}>{props.text}</Text>
          <TouchableOpacity
            style={[AppStyles.button, AppStyles.backgroundSecondary, { marginTop: 30 }]}
            onPress={() => {
              errors.setVisible(false)
            }}
          >
            <Text style={[AppStyles.h2, AppStyles.textWhite]}>Okay</Text>
          </TouchableOpacity>
        </View>
      </View>
    </>
  )
}

interface Props {
  children: React.ReactNode
}

const Errors: React.FC<Props> = ({ children }) => {
  const [visible, setVisible] = useState(false)
  const [text, setText] = useState('')
  const [path, setPath] = useState('')

  return (
    <ErrorsContext.Provider value={{ setVisible: setVisible, setText: setText, setPath: setPath }}>
      <View>
        {children}
        {visible ? (
          <>
            <ErrorDialog text={text} path={path} />
          </>
        ) : null}
      </View>
    </ErrorsContext.Provider>
  )
}

export default Errors
export { ErrorsContext }

const styles = StyleSheet.create({
  errorView: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    alignSelf: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgba(200, 200, 200, 0.78)',
  },
  errorMessage: {
    bottom: 40,
    width: '84%',
    height: '44%',
  },
})
