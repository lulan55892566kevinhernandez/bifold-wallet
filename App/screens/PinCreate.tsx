import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Alert, Keyboard } from 'react-native'
import * as Keychain from 'react-native-keychain'
import { RouteProp } from '@react-navigation/core'
import { AuthenticateStackParams } from 'navigators/AuthenticateStack'

import { Button, SafeAreaScrollView, TextInput } from 'components'

interface Props {
  route: RouteProp<AuthenticateStackParams>
}

const PinCreate: React.FC<Props> = ({ route }) => {
  const [pin, setPin] = useState('')
  const [pinTwo, setPinTwo] = useState('')
  const { t } = useTranslation()

  const passcodeCreate = async (x: string) => {
    const passcode = JSON.stringify(x)
    const description = t('user authentication pin')
    await Keychain.setGenericPassword(description, passcode, {
      service: 'passcode',
    })
  }

  const confirmEntry = (x: string, y: string) => {
    if (x.length < 6 || y.length < 6) {
      Alert.alert(t('Pin must be 6 digits in length'))
    } else if (x !== y) {
      Alert.alert(t('Pins entered do not match'))
    } else {
      passcodeCreate(x)
      route?.params?.setAuthenticated(true)
    }
  }

  return (
    <SafeAreaScrollView>
      <TextInput
        label={t('Enter Pin')}
        placeholder={t('6 Digit Pin')}
        accessible={true}
        accessibilityLabel={t('Enter Pin')}
        maxLength={6}
        autoFocus
        keyboardType="numeric"
        secureTextEntry={true}
        value={pin}
        onChangeText={setPin}
      />
      <TextInput
        label={t('Re-Enter Pin')}
        accessible={true}
        accessibilityLabel={t('Re-Enter Pin')}
        placeholder={t('6 Digit Pin')}
        maxLength={6}
        keyboardType="numeric"
        secureTextEntry
        value={pinTwo}
        onChangeText={(text: string) => {
          setPinTwo(text)
          if (text.length === 6) {
            Keyboard.dismiss()
          }
        }}
      />
      <Button
        title={t('Create')}
        accessibilityLabel={t('Create')}
        onPress={() => {
          Keyboard.dismiss()
          confirmEntry(pin, pinTwo)
        }}
      />
    </SafeAreaScrollView>
  )
}

export default PinCreate
