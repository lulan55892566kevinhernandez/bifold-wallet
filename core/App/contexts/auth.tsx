import { Agent, AgentConfig } from '@aries-framework/core'
import { IndyWallet } from '@aries-framework/core/build/wallet/IndyWallet'
import { agentDependencies } from '@aries-framework/react-native'
import React, { createContext, useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'

import {
  secretForPIN,
  storeWalletSecret,
  loadWalletSecret,
  isBiometricsActive,
  wipeWalletKey,
} from '../services/keychain'
import { WalletSecret } from '../types/security'
import { hashPIN } from '../utils/crypto'

export interface AuthContext {
  checkPIN: (pin: string) => Promise<boolean>
  getWalletCredentials: () => Promise<WalletSecret | undefined>
  setPIN: (pin: string) => Promise<void>
  commitPIN: (useBiometry: boolean) => Promise<boolean>
  isBiometricsActive: () => Promise<boolean>
}

export const AuthContext = createContext<AuthContext>(null as unknown as AuthContext)

export const AuthProvider: React.FC = ({ children }) => {
  const [walletSecret, setWalletSecret] = useState<WalletSecret>()
  const { t } = useTranslation()

  const setPIN = async (pin: string): Promise<void> => {
    const secret = await secretForPIN(pin)
    await storeWalletSecret(secret)
  }

  const commitPIN = async (useBiometry: boolean): Promise<boolean> => {
    const secret = walletSecret
    if (!secret) {
      return false
    }
    if (useBiometry) {
      await storeWalletSecret(secret, useBiometry)
    } else {
      // erase wallet key if biometrics is disabled
      await wipeWalletKey(useBiometry)
    }
    return true
  }

  const checkPIN = async (pin: string): Promise<boolean> => {
    const secret = await loadWalletSecret('', '', false)

    if (!secret || !secret.salt) {
      return false
    }

    const hash = await hashPIN(pin, secret.salt)

    try {
      await agentDependencies.indy.openWallet({ id: secret.id }, { key: hash })
      // need full secret in volatile memory in case user wants to fall back to using PIN
      const fullSecret = await secretForPIN(pin, secret.salt)
      setWalletSecret(fullSecret)
      return true
    } catch (e) {
      return false
    }
  }

  const getWalletCredentials = async (): Promise<WalletSecret | undefined> => {
    if (walletSecret) {
      return walletSecret
    }

    const secret = await loadWalletSecret(t('Biometry.UnlockPromptTitle'), t('Biometry.UnlockPromptDescription'))
    if (!secret) {
      return
    }

    setWalletSecret(secret)

    return secret
  }

  return (
    <AuthContext.Provider
      value={{
        checkPIN,
        getWalletCredentials,
        commitPIN,
        setPIN,
        isBiometricsActive,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = () => useContext(AuthContext)
