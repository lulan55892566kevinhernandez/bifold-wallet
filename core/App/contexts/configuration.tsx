import { createContext, useContext } from 'react'

import { RecordProps } from '../components/record/Record'
import OnboardingPages from '../screens/OnboardingPages'
import { Overlay } from '../types/overlay'

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  homeContentView: React.FC
  OCABundle: { branding: Record<string, Overlay>; oca: JSON | undefined }
  useBiometry: React.FC
  record: React.FC<RecordProps>
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
