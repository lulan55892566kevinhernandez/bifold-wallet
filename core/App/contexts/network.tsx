import { NetInfoStateType, useNetInfo } from '@react-native-community/netinfo'
import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import NetInfoModal from '../components/modals/NetInfoModal'

export interface NetworkContext {
  silentAssertConnectedNetwork: () => boolean
  assertConnectedNetwork: () => boolean
  displayNetInfoModal: () => void
  hideNetInfoModal: () => void
}

export const NetworkContext = createContext<NetworkContext>(null as unknown as NetworkContext)

export const NetworkProvider: React.FC = ({ children }) => {
  const netInfo = useNetInfo()
  const [isNetInfoModalDisplayed, setIsNetInfoModalDisplayed] = useState<boolean>(false)

  const displayNetInfoModal = () => {
    setIsNetInfoModalDisplayed(true)
  }

  const hideNetInfoModal = () => {
    setIsNetInfoModalDisplayed(false)
  }

  const silentAssertConnectedNetwork = () => {
    return netInfo.isConnected || netInfo.type !== NetInfoStateType.none
  }

  const assertConnectedNetwork = () => {
    const isConnected = silentAssertConnectedNetwork()
    if (!isConnected) {
      displayNetInfoModal()
    }
    return isConnected
  }

  return (
    <NetworkContext.Provider
      value={{
        silentAssertConnectedNetwork,
        assertConnectedNetwork,
        displayNetInfoModal,
        hideNetInfoModal,
      }}
    >
      {children}
      <NetInfoModal visible={isNetInfoModalDisplayed} onSubmit={() => hideNetInfoModal()} />
    </NetworkContext.Provider>
  )
}

export const useNetwork = () => useContext(NetworkContext)
