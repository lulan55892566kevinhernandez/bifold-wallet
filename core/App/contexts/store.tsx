import React, { createContext, Dispatch, useContext, useReducer } from 'react'

import { State } from '../types/state'

import reducer, { ReducerAction } from './reducers/store'

const initialState: State = {
  onboarding: {
    didAgreeToTerms: false,
    didCompleteTutorial: false,
    didCreatePIN: false,
    didConsiderBiometry: false,
  },
  authentication: {
    didAuthenticate: false,
  },
  lockout: {
    displayNotification: false,
  },
  privacy: {
    didShowCameraDisclosure: false,
  },
  preferences: {
    useBiometry: false,
  },
  credential: {
    revoked: new Set(),
    revokedMessageDismissed: new Set(),
  },
  error: null,
  loading: false,
}

export const StoreContext = createContext<[State, Dispatch<ReducerAction>]>([
  initialState,
  () => {
    return
  },
])

export const StoreProvider: React.FC = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)
  return <StoreContext.Provider value={[state, dispatch]}>{children}</StoreContext.Provider>
}

export const useStore = () => useContext(StoreContext)
