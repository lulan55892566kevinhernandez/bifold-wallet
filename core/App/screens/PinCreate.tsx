import { useNavigation } from '@react-navigation/core'
import { StackNavigationProp } from '@react-navigation/stack'
import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Platform, Keyboard, StyleSheet, Text, StatusBar, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import Button, { ButtonType } from '../components/buttons/Button'
import PinInput from '../components/inputs/PinInput'
import AlertModal from '../components/modals/AlertModal'
import { minPINLength } from '../constants'
import { useAuth } from '../contexts/auth'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import { AuthenticateStackParams, Screens } from '../types/navigators'
import { statusBarStyleForColor, StatusBarStyles } from '../utils/luminance'
import { testIdWithKey } from '../utils/testable'

interface PinCreateProps {
  setAuthenticated: (status: boolean) => void
}

interface ModalState {
  visible: boolean
  title: string
  message: string
}

const PinCreate: React.FC<PinCreateProps> = ({ setAuthenticated }) => {
  const { setPIN } = useAuth()
  const [pin, setPin] = useState('')
  const [pinTwo, setPinTwo] = useState('')
  const [continueEnabled, setContinueEnabled] = useState(true)
  const [modalState, setModalState] = useState<ModalState>({
    visible: false,
    title: '',
    message: '',
  })
  const navigation = useNavigation<StackNavigationProp<AuthenticateStackParams>>()
  const [, dispatch] = useStore()
  const { t } = useTranslation()

  const { ColorPallet, TextTheme } = useTheme()
  const style = StyleSheet.create({
    container: {
      height: '100%',
      backgroundColor: ColorPallet.brand.primaryBackground,
      padding: 20,
    },
  })

  const passcodeCreate = async (pin: string) => {
    // TODO: Update this
    try {
      setContinueEnabled(false)
      await setPIN(pin)
      // This will trigger initAgent
      setAuthenticated(true)

      dispatch({
        type: DispatchAction.DID_CREATE_PIN,
      })

      // TODO: Navigate back if in settings
      navigation.navigate(Screens.UseBiometry)
    } catch (e) {
      // TODO:(jl)
    }
  }

  const confirmEntry = async (x: string, y: string) => {
    const negativePattern = /[^0-9]/g
    if (negativePattern.test(x)) {
      setModalState({
        visible: true,
        title: t('PinCreate.InvalidPIN'),
        message: t('PinCreate.PleaseUseOnlyNumbersInYourPIN'),
      })
    } else if (!x.length) {
      setModalState({
        visible: true,
        title: t('PinCreate.EnterPINTitle'),
        message: t('PinCreate.YouNeedToCreateA6DigitPIN'),
      })
    } else if (x.length < minPINLength) {
      setModalState({
        visible: true,
        title: t('PinCreate.PINTooShort'),
        message: t('PinCreate.YourPINMustBe6DigitsInLength'),
      })
    } else if (negativePattern.test(y)) {
      setModalState({
        visible: true,
        title: t('PinCreate.InvalidPIN'),
        message: t('PinCreate.PleaseUseOnlyNumbersInYourPIN'),
      })
    } else if (!y.length) {
      setModalState({
        visible: true,
        title: t('PinCreate.ReenterPINTitle'),
        message: t('PinCreate.PleaseReenterYourPIN'),
      })
    } else if (y.length < minPINLength) {
      setModalState({
        visible: true,
        title: t('PinCreate.PINTooShort'),
        message: t('PinCreate.YourPINMustBe6DigitsInLength'),
      })
    } else if (x !== y) {
      setModalState({
        visible: true,
        title: t('PinCreate.PINsDoNotMatch'),
        message: t('PinCreate.EnteredPINsDoNotMatch'),
      })
    } else {
      await passcodeCreate(x)
    }
  }

  return (
    <SafeAreaView>
      <StatusBar
        barStyle={
          Platform.OS === 'android' ? StatusBarStyles.Light : statusBarStyleForColor(style.container.backgroundColor)
        }
      />
      <View style={[style.container]}>
        <Text style={[TextTheme.normal, { marginBottom: 16 }]}>
          <Text style={{ fontWeight: 'bold' }}>{t('PinCreate.RememberPIN')}</Text> {t('PinCreate.PINDisclaimer')}
        </Text>
        <PinInput
          label={t('PinCreate.EnterPINTitle')}
          onPinChanged={setPin}
          testID={testIdWithKey('EnterPIN')}
          accessibilityLabel={t('PinCreate.EnterPIN')}
          autoFocus={true}
        />
        <PinInput
          label={t('PinCreate.ReenterPIN')}
          onPinChanged={(p: string) => {
            setPinTwo(p)
            if (p.length === minPINLength) {
              Keyboard.dismiss()
            }
          }}
          testID={testIdWithKey('ReenterPIN')}
          accessibilityLabel={t('PinCreate.ReenterPIN')}
        />
        {modalState.visible && (
          <AlertModal
            title={modalState.title}
            message={modalState.message}
            submit={() => setModalState({ ...modalState, visible: false })}
          />
        )}
      </View>
      <View style={{ marginTop: 'auto', margin: 20 }}>
        <Button
          title={t('PinCreate.CreatePIN')}
          testID={testIdWithKey('CreatePIN')}
          accessibilityLabel={t('PinCreate.CreatePIN')}
          buttonType={ButtonType.Primary}
          disabled={!continueEnabled}
          onPress={async () => {
            Keyboard.dismiss()
            await confirmEntry(pin, pinTwo)
          }}
        />
      </View>
    </SafeAreaView>
  )
}

export default PinCreate
