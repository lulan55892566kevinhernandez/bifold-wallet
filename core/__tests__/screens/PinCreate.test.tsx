import { render } from '@testing-library/react-native'
import React from 'react'

import { AuthContext } from '../../App/contexts/auth'
import PinCreate from '../../App/screens/PinCreate'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

jest.mock('react-native-device-info', () => {
  return {
    isEmulatorSync: () => {
      return true
    },
  }
})

describe('displays a pin create screen', () => {
  test.skip('pin create renders correctly', () => {
    // const tree = render(
    //   <AuthContext.Provider
    //     value={{
    //       getWalletCredentials: jest.fn(),
    //       checkPIN: jest.fn(),
    //       commitPIN: jest.fn(),
    //       setPIN: jest.fn(),
    //       isBiometricsActive: jest.fn(),
    //     }}
    //   >
    //     <PinCreate setAuthenticated={jest.fn()} />
    //   </AuthContext.Provider>
    // )
    expect(tree).toMatchSnapshot()
  })
})
