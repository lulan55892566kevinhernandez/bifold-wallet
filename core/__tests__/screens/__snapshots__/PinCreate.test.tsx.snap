// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a pin create screen pin create renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "margin": 20,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Object {
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      PinCreate.RememberPIN
    </Text>
     
    PinCreate.PINDisclaimer
  </Text>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          Object {
            "marginBottom": 8,
          },
        ]
      }
    >
      PinCreate.EnterPIN
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
            Object {
              "marginBottom": 24,
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.EnterPIN-1"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="EnterPIN-1"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.EnterPIN-2"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="EnterPIN-2"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.EnterPIN-3"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="EnterPIN-3"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.EnterPIN-4"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="EnterPIN-4"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.EnterPIN-5"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="EnterPIN-5"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.EnterPIN-6"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="EnterPIN-6"
          >
            
          </Text>
        </View>
        <TextInput
          autoCapitalize="characters"
          autoCorrect={false}
          autoFocus={true}
          blurOnSubmit={false}
          caretHidden={true}
          clearButtonMode="never"
          disableFullscreenUI={true}
          keyboardType="numeric"
          maxLength={6}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onPressOut={[Function]}
          spellCheck={false}
          style={
            Object {
              "bottom": 0,
              "fontSize": 1,
              "left": 0,
              "opacity": 0.01,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="EnterPIN"
          textContentType="password"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginBottom": 32,
            "marginRight": 8,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î£´
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          Object {
            "marginBottom": 8,
          },
        ]
      }
    >
      PinCreate.ReenterPIN
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
            Object {
              "marginBottom": 24,
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.ReenterPIN-1"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="ReenterPIN-1"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.ReenterPIN-2"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="ReenterPIN-2"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.ReenterPIN-3"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="ReenterPIN-3"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.ReenterPIN-4"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="ReenterPIN-4"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.ReenterPIN-5"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="ReenterPIN-5"
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "borderColor": "#313132",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 48,
                "marginRight": 8,
                "width": 40,
              },
              false,
            ]
          }
        >
          <Text
            accessibilityLabel="PinCreate.ReenterPIN-6"
            accessible={true}
            maxFontSizeMultiplier={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "lineHeight": 42,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
            testID="ReenterPIN-6"
          >
            
          </Text>
        </View>
        <TextInput
          autoCapitalize="characters"
          autoCorrect={false}
          autoFocus={false}
          blurOnSubmit={false}
          caretHidden={true}
          clearButtonMode="never"
          disableFullscreenUI={true}
          keyboardType="numeric"
          maxLength={6}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onPressOut={[Function]}
          spellCheck={false}
          style={
            Object {
              "bottom": 0,
              "fontSize": 1,
              "left": 0,
              "opacity": 0.01,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="ReenterPIN"
          textContentType="password"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginBottom": 32,
            "marginRight": 8,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î£´
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="PinCreate.CreatePIN"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#42803E",
        "borderRadius": 4,
        "opacity": 1,
        "padding": 16,
      }
    }
    testID="com.ariesbifold:id/CreatePIN"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          false,
        ]
      }
    >
      PinCreate.CreatePIN
    </Text>
  </View>
</RNCSafeAreaView>
`;
